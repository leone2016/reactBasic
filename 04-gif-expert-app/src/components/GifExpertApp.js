import React, { useState } from 'react'
import AddCategory from './AddCategory';
import { GifGrid } from './GifGrid';

const GifExpertApp = props => {
    const initialState = ['One Punch'];
    const [categories, setcategories] = useState(initialState);

    const handleAdd = (category) => {
        /* setcategories( ['HunterHunter', ...categories]); */
    }
    return (
        <div>
            <h2>GifExpertApp</h2>
            <AddCategory setcategories={setcategories} />
            <hr />
            <button onClick={handleAdd}>Agregar</button>
            <ol>
                {
                    categories.map((cat) => (
                      <GifGrid
                      key={cat}
                      category={cat}
                      />
                    )

                    )
                }
            </ol>
        </div>
    )
}

export default GifExpertApp
