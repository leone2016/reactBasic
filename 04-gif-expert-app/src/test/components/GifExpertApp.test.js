import React from 'react';
import { shallow } from "enzyme";
import '@testing-library/jest-dom';
import GifExpertApp from "../../components/GifExpertApp";

describe('UNIT TEST GifExpertApp.test.js', () => {
    test('should be show <GifExpertApp/> correctly', () => {
        const wrapper = shallow(<GifExpertApp/>);
        expect(wrapper).toMatchSnapshot();
        // el estado al no tener un identificador useState en componentes se visualiza como State: ['One Punch']
        // ya que se podria tener varios estados
        //  una forma para validar es que los valores iniciales vengan por las props
        // de esta manera no se envia ONE PUNCH por efecto
    });

    test('should be show a list of categories', ()=>{
        const categories = ['One Punch', 'Dragon Ball']; // esta es una forma de mokear el useState ( )

        const wrapper = shallow(<GifExpertApp defaultCategories={categories}/>);
        expect(wrapper).toMatchSnapshot();
        expect( wrapper.find('GifGrid').length).toBe(categories.length); // las veces que se imprima GifGrid es igual a categories


    });
    
});
