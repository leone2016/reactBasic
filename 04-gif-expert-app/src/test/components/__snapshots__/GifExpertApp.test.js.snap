// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UNIT TEST GifExpertApp.test.js should be show <GifExpertApp/> correctly 1`] = `
<div>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setcategories={[Function]}
  />
  <hr />
  <button
    onClick={[Function]}
  >
    Agregar
  </button>
  <ol />
</div>
`;

exports[`UNIT TEST GifExpertApp.test.js should be show a list of categories 1`] = `
<div>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setcategories={[Function]}
  />
  <hr />
  <button
    onClick={[Function]}
  >
    Agregar
  </button>
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ol>
</div>
`;
